dnsmasq_interface.c

```diff
@@ -266,27 +288,30 @@ void FTL_forwarded(unsigned int flags, char *name, struct all_addr *addr, int id
 	// Proceed only if
 	// - current query has not been marked as replied to so far
 	//   (it could be that answers from multiple forward
-	//    destionations are coimg in for the same query)
+	//    destination are coming in for the same query)
 	// - the query was formally known as cached but had to be forwarded
 	//   (this is a special case further described below)
-	if(queries[i].complete && queries[i].status != QUERY_CACHE)
+	if(queries[queryID].complete && queries[queryID].status != QUERY_CACHE)
 	{
 		free(forward);
 		disable_thread_lock();
 		return;
 	}
 
+	// Save fate of this query (it got forwarded)
+	queries[queryID].status = QUERY_FORWARDED;
+
 	// Get ID of forward destination, create new forward destination record
 	// if not found in current data structure
 	int forwardID = findForwardID(forward, true);
-	queries[i].forwardID = forwardID;
+	queries[queryID].forwardID = forwardID;
 
-	if(!queries[i].complete)
+	if(!queries[queryID].complete)
 	{
-		int j = queries[i].timeidx;
+		int j = queries[queryID].timeidx;
 		validate_access("overTime", j, true, __LINE__, __FUNCTION__, __FILE__);
 
-		if(queries[i].status == QUERY_CACHE)
+		if(queries[queryID].status == QUERY_CACHE)
```

Comment:
If the status is always set to `QUERY_FORWARDED` on line 301, then wouldn't this check always fail?